### PR TITLE
feat(all): swap error names to AOT compile-safe comparison

### PR DESCRIPTION
**Pull request by damienwebdev**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?
Currently, we swap on function names on errors, turns out this doesn't actually work in AOT builds, as names are mutilated by the compiler. 

Fixes: N/A

## What is the new behavior?
This introduces a static "code" property to Daffodil errors that we can switch against reliably. 

## Does this PR introduce a breaking change?
```
[x] Yes
[ ] No
```

This change is breaking, but more of a "fix" as the original code never really functioned correctly.

## Other information

### CODE REVIEW

File: libs/cart/src/effects/cart-address.effects.ts [hunk 1]
```diff
@@ -32,7 +32,7 @@ export class DaffCartAddressEffects<T extends DaffCartAddress, V extends DaffCar
       map(() => this.storage.getCartId()),
       switchMap(cartId => this.driver.update(cartId, action.payload)),
       map((resp: V) => new DaffCartAddressUpdateSuccess(resp)),
-      catchError(error => of(error.name === DaffStorageServiceError.name
+      catchError(error => of(error.code === DaffStorageServiceError.code
```
Comment (griest024) [inline libs/cart/src/effects/cart-address.effects.ts:35]: `code` is a class field so you would need to do something like `error.constructor.code`.

**Review by griest024 — CHANGES_REQUESTED** [review]

File: libs/cart/src/effects/cart-address.effects.ts (hunk 1 above)
Comment (damienwebdev) [reply libs/cart/src/effects/cart-address.effects.ts:35]: I'm not 100% certain, but I believe typescript compilation may do this automatically.

I'll test tomorrow and verify.

Comment (damienwebdev) [reply libs/cart/src/effects/cart-address.effects.ts]: I've swapped this for instanceof checking.

File: libs/core/src/errors/inheritable-error.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+ * 
+ * let myError = new MyError();
+ * 
+ * myError instanceof MyError; // returns false
+ * ```
+ * 
+ * You will see unexpected things. This class fixes that issue as described here
+ * https://github.com/microsoft/TypeScript/issues/13965
+ */
+export class DaffInheritableError extends Error {
```
Comment (griest024) [inline libs/core/src/errors/inheritable-error.ts:18]: Can you add an integration test demonstrating that inheritance checking works?

**Review by griest024 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**